solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -835,13 +858,19 @@ pub struct AddUserAuthorityDelegate<'info> {
     #[account(
         init,
         payer = payer,
-        seeds = [&user.key().to_bytes()[..32], &user_authority_delegate.to_bytes()[..32]],
+        seeds = [&user.key().to_bytes()[..32], &delegate_pubkey.to_bytes()[..32]],
         bump,
         space = USER_AUTHORITY_DELEGATE_ACCOUNT_SIZE
     )]
-    pub user_authority_delegate_pda: Account<'info, UserAuthorityDelegate>,
+    pub new_user_authority_delegate: Account<'info, UserAuthorityDelegate>,
+    /// CHECK: Signer/user delegate, can be defaulted to SystemProgram for no-op
     #[account()]
-    pub user_authority: Signer<'info>,
+    pub signer_user_authority_delegate: AccountInfo<'info>, // provided for authority authentication
+    /// CHECK: Signer's delegation status, can be defaulted to SystemProgram for no-op
```

Comment:
this is to prevent anchor build errors from passing in raw AccountInfo - https://book.anchor-lang.com/chapter_3/the_accounts_struct.html#safety-checks